queue.c

```diff
@@ -85,11 +85,14 @@ queue_empty(queue_t *queue) {
 }
 
 void
-queue_submit(queue_t *queue, queue_item_t *item) {
+queue_submit(queue_t *queue, void *data) {
     if (queue == NULL) {
         return;
     }
 
+    // TODO(jgerth): Error check create function.
```

Comment:
This `TODO` should be resolved before this can be merged.

Reply:
There are two (and a combined third) options:

1. Return an error value from `queue_submit()` if `queue_item_create()` fails (e.g. `0` on success and `!= 0` on error)
2. Internally retry `queue_item_create()` until successful
3. **2.**, but limit the number of retries, returning an error value as described in **1.**, if `queue_item_create()` still fails on the last possible retry

Reply:
Went with option **2.** here for now, as it is the simplest to implement and does not change the external API of the function. Resolving...